### ISSUE TITLE
php7 missing modules?

### ISSUE THREAD

**Issue by alecpl** [oerdnj/deb.sury.org#95]
I installed php7.0 but it contains only a few modules.
```
php -m
[PHP Modules]
Core
date
hash
libxml
mysqlnd
pcntl
pcre
PDO
phpdbg_webhelper
Reflection
SPL
standard
```
and I can't find a way to install others like dom, json, etc. apt-get instrall php-json does not find a package. I've got php 5.5 before, uninstalled it and disabled the repo. What am I missing here?

**Comment by oerdnj** [comment]
There's a prominent sign everywhere, that the repository is still very much experimental...

Building the modules on trusty is still broken, the modules will be available when I solve the building problem.

**Comment by markc** [comment]
I probably should have opened a new ticket but it is releated to php7 and modules. Anyway this is really just a FWIW/FYI...
```
~ php phpinfo.php
Failed loading /usr/lib/php/20141001/opcache.so:  /usr/lib/php/20141001/opcache.so: undefined symbol: pcre_free
/home/u/gc5/var/www/phpinfo.php
Segmentation fault (core dumped)
~ dpkg -S /usr/lib/php/20141001/opcache.so
php-opcache: /usr/lib/php/20141001/opcache.so
```

**Comment by oerdnj** [comment]
Please recheck with PHP 7.0.0~beta3 build, but only after php-modules 9 are built ok (which might take a while).

**Comment by alecpl** [comment]
Much better in CLI, but apache restart gives me: Syntax error on line 1 of /etc/apache2/mods-enabled/php7.0.load: Can't locate API module structure `php7.0_module' in file /usr/lib/apache2/modules/libphp7.0.so: /usr/lib/apache2/modules/libphp7.0.so: undefined symbol: php7.0_module

**Comment by alecpl** [comment]
I also miss openssl extension.

**Comment by markc** [comment]
I suspect it's a bit early yet for all extension but I could use ssh2.

And thanks for all your efforts BTW. It's wonderful to be able to easily test php7 at this early stage.

**Comment by alecpl** [comment]
Oh, and another essential extension missing: 'sockets'.

**Comment by oerdnj** [comment]
I have added gettext, openssl and sockets in the latest build.

The segfaults will probably cease closer to the full release.

Unfortunately the build infrastructure is not yet ready for building other modules; porting php5.6 and php5.5 to the new build system has higher priority.

**Comment by alecpl** [comment]
Thank you for adding missing modules. Unfortunately the apache module does not work. Also, I see that composer can't install some PEAR packages because of some phar+tar issue. So, at the moment I can't do much testing.

**Comment by oerdnj** [comment]
@alecpl 
> apache restart gives me: Syntax error on line 1 of /etc/apache2/mods-enabled/php7.0.load: Can't locate API module structure `php7.0_module' in file /usr/lib/apache2/modules/libphp7.0.so: /usr/lib/apache2/modules/libphp7.0.so: undefined symbol: php7.0_module

The load file should not contain php7_module, so hopefully this has been fixed as well (I haven't tested it though yet).

**Comment by alecpl** [comment]
Setting up libapache2-mod-php7.0 (7.0.0~beta3-4+deb.sury.org~trusty+1) ...
Installing new version of config file /etc/apache2/mods-available/php7.0.load ...
/var/lib/dpkg/info/libapache2-mod-php7.0.postinst: 62: /var/lib/dpkg/info/libapache2-mod-php7.0.postinst: php_enable: not found
Setting up php7.0 (7.0.0~beta3-4+deb.sury.org~trusty+1) ...
Setting up ssh (1:6.6p1-2ubuntu2.2) ...
Errors were encountered while processing:
 php7.0-cgi
E: Sub-process /usr/bin/dpkg returned an error code (1)

and on apache start now:

apache2: Syntax error on line 140 of /etc/apache2/apache2.conf: Syntax error on line 1 of /etc/apache2/mods-enabled/php7.0.load: Can't locate API module structure `php@PHP_MAJOR_VERSION@_module' in file /usr/lib/apache2/modules/libphp7.0.so: /usr/lib/apache2/modules/libphp7.0.so: undefined symbol: php@PHP_MAJOR_VERSION@_module
Action 'configtest' failed.

I fixed manually the php7.0.load file and now I got it working. Thanks.

**Comment by erikstainsby** [comment]
@alecpl Would you post the content of the php7.load file with the correct symbol in it? I keep seeing:  `/usr/lib/libphp7.so: undefined symbol: php7_module` when I try to test apache2ctl
